projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -487,7 +501,7 @@ public static function user_can_view_post() {
 
 		require_once JETPACK__PLUGIN_DIR . 'extensions/blocks/premium-content/_inc/subscription-service/include.php';
 		$paywall       = \Automattic\Jetpack\Extensions\Premium_Content\subscription_service();
-		$can_view_post = $paywall->visitor_can_view_content( self::get_all_plans_id_jetpack_recurring_payments(), $post_access_level );
+		$can_view_post = $paywall->visitor_can_view_content( self::get_all_plans_id_jetpack_recurring_payments( self::$plan_type_newsletter ), $post_access_level );
```

Comment:
What is your idea behind adding this parameter? get_all_plans_id_jetpack_recurring_payments() seems to only be used for Newsletters (Premium content block is using a different approach: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Qcyhtva%2Sfgntvat%2Srkgrafvbaf%2Soybpxf%2Scerzvhz%2Qpbagrag%2S_vap%2Snpprff%2Qpurpx.cuc%3Se%3Qno0p29np%2385-og by getting the plan from the Block context.

The way you implemented it is fine but adds some complexity when it is not needed I think.

What about renaming `get_all_plans_id_jetpack_recurring_payments` to something more meaningful in the NL context instead?

Keep in mind we will revisit this as well once we unlock the different user tiers...

Reply:
> What is your idea behind adding this parameter? get_all_plans_id_jetpack_recurring_payments() seems to only be used for Newsletters (Premium content block is using a different approach:

@millerf I thought premium content was using non-newsletter plans, like [here](https://github.com/Automattic/jetpack/pull/31450/files#diff-d31988b4a8aab0212b66d79f7ff48aa585b26d40f59dabfe642f3be01c0cc47aL39). After reading the fn comment I see "newsletter" being mentioned but the function is in the premium-content folder

Reply:
> I thought premium content was using non-newsletter plans, like here


Two things here:

First of all, I made a mistake, and I did not see your use of `plan_type_non_newsletter`. 
I did not see it because 
[class-wpcom-offline-subscription-service.php](https://github.com/Automattic/jetpack/pull/31450/files#diff-d31988b4a8aab0212b66d79f7ff48aa585b26d40f59dabfe642f3be01c0cc47a) actually gates sending newsletter (see here: fbhepr%2Skers%2Sjcpbz%2Snflap%2Qwbof%2Svapyhqrf%2Sfhofpevcgvba%2Qrznvyf%2Sfhofpevcgvba%2Qrznvyf.cuc%3Se%3Q123p982o%23406-og

But you are right, Premium content uses non-newsletters plans.
 
My understanding about `Premium_Content\subscription_service()` is that it is actually only gating Newsletters (both access to post and email sending). 
The different classes (Offline, Online, Token) only depends on the environment (Async jobs, WPCOM or Jetpack  - in that order, I think). 

That's my understanding, I might be wrong...

Reply:
> but the function is in the premium-content folder

Yes, I am not exactly sure why it is there.

Reply:
From the history:

https://github.com/Automattic/jetpack/commit/94ddae8284758db97529fad30752b9e11a556bb1#diff-bbd4ff6d14a8af458b4095fb0207187272efbf16d8e29301714d2a5c452c40e2L94

It looks like we piggy backed on the gating of Premium content block and extended it:

<img width="629" alt="Screenshot 2023-06-20 at 16 42 00" src="https://github.com/Automattic/jetpack/assets/790558/46f089d9-b99e-47b4-acbf-4ecf6d2d3ff8">

So it seems all the different classes were here already (except `Online`).

Now, you are right, there must be some gating on WPCOM when sending emails as well for Premium content...

Reply:
Fixed in [f8cae35](https://github.com/Automattic/jetpack/pull/31450/commits/f8cae3533d66378c88f64d544371a4e6956d7903)